packages/thrift-server-core/src/main/protocols/TProtocol.ts

```diff
@@ -75,7 +76,7 @@ export abstract class TProtocol {
 
     public abstract writeI32(i32: number): void
 
-    public abstract writeI64(i64: number | Int64): void
+    public abstract writeI64(i64: number | string | IInt64): void
```

Comment:
The implementations were all checking for string types. Add string to the parameter type.